xds/internal/resolver/watch_service.go

```diff
@@ -71,22 +71,22 @@ func newRouteConfigWatcher(resourceName string, parent *xdsResolver) *routeConfi
 	return rw
 }
 
-func (r *routeConfigWatcher) OnUpdate(update *xdsresource.RouteConfigResourceData) {
-	r.parent.serializer.TrySchedule(func(context.Context) {
-		r.parent.onRouteConfigResourceUpdate(r.resourceName, update.Resource)
-	})
+func (r *routeConfigWatcher) OnUpdate(u *xdsresource.RouteConfigResourceData, onDone xdsresource.DoneNotifier) {
+	handleUpdate := func(context.Context) {
```

Comment:
Optional: no comment here about ensuring the on done callback is always called. I prefer it like this since it's implicit from the code. This comment is present in other watchers.

Reply:
Hmm ... some places are a little more involved than others, as in, the update might not be processed inline or directly via a serializer (as is the case here). So, I felt the need to add comments in some places. If you feel its too verbose in certain places, do let me know. Thanks.

Reply:
Some of the watchers that have this comment are exactly like this with scheduleOr. So I'd prefer to remove the comment from those specifically since it's implicit from the call to ScheduleOr.